components/chart/src/chart.ts

```diff
@@ -671,10 +685,10 @@ export class DtChart
   /** Checks if a heatfield is supported with the chart options if not throw an error */
   private _checkHeatfieldSupport(): void {
     if (
-      this.options &&
-      this.options.xAxis &&
-      this.options.xAxis[0] &&
-      this.options.xAxis[0].categories
+      this._currentOptions &&
+      this._currentOptions.xAxis &&
+      this._currentOptions.xAxis[0] &&
+      this._currentOptions.xAxis[0].categories
```

Comment:
NIT: since we have typescript 3.7 I would suggest using optional chaining.

```suggestion
      this._currentOptions?.xAxis?[0]?.categories
```